tasks/parallel_verification.py

```diff
@@ -0,0 +1,183 @@
+                ),
+            )
+        if not chunks_legacy:
+            log.info(
+                "Legacy chunks did not match parallel results",
+                extra=dict(
+                    repoid=repoid,
+                    commitid=commitid,
+                    commit_yaml=commit_yaml,
+                    report_code=report_code,
+                    parallel_paths=parallel_paths,
+                ),
+            )
+        if not chunks_regular:
+            log.info(
+                "Chunks did not match parallel results",
+                extra=dict(
+                    repoid=repoid,
+                    commitid=commitid,
+                    commit_yaml=commit_yaml,
+                    report_code=report_code,
+                    parallel_paths=parallel_paths,
+                ),
+            )
+
+        verification_result = (
+            (1 if fas_legacy else 0)
+            + (1 if fas_regular else 0)
+            + (1 if chunks_legacy else 0)
+            + (1 if chunks_regular else 0)
+        ) / 4
+
+        if verification_result == 1:
+            log.info(
+                "Parallel upload processsing verification succeeded",
+                extra=dict(
+                    repoid=repoid,
+                    commitid=commitid,
+                    commit_yaml=commit_yaml,
+                    report_code=report_code,
+                    parallel_paths=parallel_paths,
+                ),
+            )
+        else:
+            log.info(
+                f"Parallel upload processsing verification failed with {verification_result}",
+                extra=dict(
+                    repoid=repoid,
+                    commitid=commitid,
+                    commit_yaml=commit_yaml,
+                    report_code=report_code,
+                    parallel_paths=parallel_paths,
+                ),
+            )
+
+        return
+
+
+# To filter out values not relevant for verifying report content correctness. We
+# don't want these values to be the reason why there exists a diff, since it's
+# not actually related to coverage report contents
+def sort_and_stringify_report_json(d):
+    for session_id in d["sessions"].keys():
+        d["sessions"][session_id].pop("d", None)  # remove timestamp
+        d["sessions"][session_id].pop("e", None)  # remove env
+        d["sessions"][session_id].pop("p", None)  # remove state
+        d["sessions"][session_id].pop("se", None)  # remove session extras
+
+        # round the coverage precentage to 2 decimals (because we're inconsistent somewhere)
+        if "t" in d["sessions"][session_id]:
+            d["sessions"][session_id]["t"][5] = str(
+                round(float(d["sessions"][session_id]["t"][5]), 2)
+            )
+
+    # prune out totals since they're calculated deterministically from report contents
+    for file_name in d["files"].keys():
+        d["files"][file_name] = d["files"][file_name][:3]
```

Comment:
would we not expect file totals to be identical?

Reply:
we would... tbh I haven't figured out how to recalculate it yet

Reply:
figured it out!